src/core/components/HoverIntent/index.js

```diff
@@ -0,0 +1,76 @@
+import React, { PropTypes } from 'react';
+
+export default class HoverIntent extends React.Component {
+  static propTypes = {
+    sensitivity: PropTypes.number,
+    interval: PropTypes.number,
+    onHoverIntent: PropTypes.func.isRequired,
+    onHoverIntentEnd: PropTypes.func.isRequired,
+    children: PropTypes.element.isRequired,
+  }
+
+  static defaultProps = {
+    sensitivity: 5,
+    interval: 100,
+  }
+
+  componentWillUnmount() {
+    this.isHoverIntended = false;
+    this.clearHoverIntentDetection();
+  }
+
+  onMouseOver = (e) => {
+    e.persist();
+    const currentTarget = e.currentTarget;
+
+    const sq = (x) => x * x;
+    const distanceSq = (p1, p2) => sq(p1.x - p2.x) + sq(p1.y - p2.y);
+
+    const sensitivitySq = sq(this.props.sensitivity);
+
+    const initialPosition = { x: e.clientX, y: e.clientY };
+    let previousPosition = initialPosition;
+    this.currentMousePosition = initialPosition;
+
+    this.interval = setInterval(() => {
+      const currentPosition = this.currentMousePosition;
```

Comment:
So, it turns out sinon has methods for faking timers. I went with those because it made it simpler to keep the coverage at 100%. Does that seem reasonable?